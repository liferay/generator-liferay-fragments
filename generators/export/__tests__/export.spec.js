const getSiteApi = require('../../../utils/get-site-api-mock');
const getProjectContent = require('../../../utils/get-project-content');
const getTestFixtures = require('../../../utils/get-test-fixtures');
const exportCollections = require('../export');

/**
 * @type string
 */
const GROUP_ID = '1234';

describe('export-generator/export', () => {
  getTestFixtures().forEach(projectPath => {
    it('exports fragments from a Liferay Site', async () => {
      const projectContent = getProjectContent(projectPath);
      const siteApi = getSiteApi(projectContent);

      const newProjectContent = await exportCollections(
        siteApi,
        GROUP_ID,
        projectContent
      );

      expect(newProjectContent).toMatchSnapshot();
    });
  });
});
