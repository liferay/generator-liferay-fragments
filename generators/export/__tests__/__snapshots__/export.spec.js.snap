// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-generator/export exports fragments from a Liferay Site 1`] = `
Array [
  Object {
    "fragmentCollectionId": "collection-a",
    "fragmentCompositions": Array [],
    "fragments": Array [
      Object {
        "configuration": "{\\"fieldSets\\": []}",
        "css": ".fragment {}",
        "html": "<fragment-a></fragment-a>",
        "js": "console.log(\\"fragment\\")",
        "metadata": Object {
          "configurationPath": "configuration.json",
          "cssPath": "styles.css",
          "htmlPath": "index.html",
          "jsPath": "main.js",
          "name": "Fragment A",
          "type": "1",
        },
        "slug": "1234-collection-a-fragment-a",
      },
    ],
    "metadata": Object {
      "description": "This is collection A",
      "name": "Collection A",
    },
    "slug": "collection-a",
  },
  Object {
    "fragmentCollectionId": "collection-b",
    "fragmentCompositions": Array [],
    "fragments": Array [
      Object {
        "configuration": "{\\"fieldSets\\": []}",
        "css": ".fragment {}",
        "html": "<fragment-a></fragment-a>",
        "js": "console.log(\\"fragment\\")",
        "metadata": Object {
          "configurationPath": "configuration.json",
          "cssPath": "styles.css",
          "htmlPath": "index.html",
          "jsPath": "main.js",
          "name": "Fragment A",
          "type": "1",
        },
        "slug": "1234-collection-b-fragment-a",
      },
    ],
    "metadata": Object {
      "description": "This is collection B",
      "name": "Collection B",
    },
    "slug": "collection-b",
  },
]
`;

exports[`export-generator/export exports fragments from a Liferay Site 2`] = `
Array [
  Object {
    "fragmentCollectionId": "collection-a",
    "fragmentCompositions": Array [],
    "fragments": Array [
      Object {
        "configuration": "{\\"fieldSets\\": []}",
        "css": ".fragment {}",
        "html": "<fragment-a></fragment-a>",
        "js": "console.log(\\"fragment\\")",
        "metadata": Object {
          "configurationPath": "configuration.json",
          "cssPath": "styles.css",
          "htmlPath": "index.html",
          "jsPath": "main.js",
          "name": "Fragment A",
          "type": "1",
        },
        "slug": "1234-collection-a-fragment-a",
      },
    ],
    "metadata": Object {
      "description": "This is collection A",
      "name": "Collection A",
    },
    "slug": "collection-a",
  },
  Object {
    "fragmentCollectionId": "collection-b",
    "fragmentCompositions": Array [],
    "fragments": Array [
      Object {
        "configuration": "{\\"fieldSets\\": []}",
        "css": ".fragment {}",
        "html": "<fragment-a></fragment-a>",
        "js": "console.log(\\"fragment\\")",
        "metadata": Object {
          "configurationPath": "configuration.json",
          "cssPath": "styles.css",
          "htmlPath": "index.html",
          "jsPath": "main.js",
          "name": "Fragment A",
          "type": "1",
        },
        "slug": "1234-collection-b-fragment-a",
      },
    ],
    "metadata": Object {
      "description": "This is collection B",
      "name": "Collection B",
    },
    "slug": "collection-b",
  },
]
`;

exports[`export-generator/export exports fragments from a Liferay Site 3`] = `
Array [
  Object {
    "fragmentCollectionId": "collection-a",
    "fragmentCompositions": Array [],
    "fragments": Array [
      Object {
        "configuration": "{\\"fieldSets\\": []}",
        "css": ".fragment {}",
        "html": "<fragment-a></fragment-a>",
        "js": "console.log(\\"fragment\\")",
        "metadata": Object {
          "configurationPath": "configuration.json",
          "cssPath": "styles.css",
          "htmlPath": "index.html",
          "jsPath": "main.js",
          "name": "Fragment A",
          "type": "1",
        },
        "slug": "1234-collection-a-fragment-a",
      },
    ],
    "metadata": Object {
      "description": "This is collection A",
      "name": "Collection A",
    },
    "slug": "collection-a",
  },
  Object {
    "fragmentCollectionId": "collection-b",
    "fragmentCompositions": Array [],
    "fragments": Array [
      Object {
        "configuration": "{\\"fieldSets\\": []}",
        "css": ".fragment {}",
        "html": "<fragment-a></fragment-a>",
        "js": "console.log(\\"fragment\\")",
        "metadata": Object {
          "configurationPath": "configuration.json",
          "cssPath": "styles.css",
          "htmlPath": "index.html",
          "jsPath": "main.js",
          "name": "Fragment A",
          "type": "1",
        },
        "slug": "1234-collection-b-fragment-a",
      },
    ],
    "metadata": Object {
      "description": "This is collection B",
      "name": "Collection B",
    },
    "slug": "collection-b",
  },
]
`;
